aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -185,9 +193,16 @@ std::tuple<Tensor &,Tensor &> _min_out_cpu(Tensor& min, Tensor& min_indices,
 }
 
 std::tuple<Tensor, Tensor> min(const Tensor& self, int64_t dim, bool keepdim) {
-  Tensor min = at::empty({0}, self.options());
   Tensor min_indices = at::empty({0}, self.options().dtype(kLong));
-  return at::native::min_out(min, min_indices, self, dim, keepdim);
+  if (self.is_quantized()) {
```

Comment:
why put it in code instead of dispatch in native_functions.yaml? (doing so is safe as the backwards is attached at the same level)

Reply:
right now min in native_functions.yaml is abstract function which does not have any dispatch, should I add dispatch to it? cc @gchanan?

Reply:
I think the general pattern we should enforce is if it should be dispatched, do it in native_functions.yaml

Reply:
@li-roy I think that's close to the opposite of the pattern we want.  The issue with that (besides breaking backwards definitions), is you have to repeat all the error checking code and it's very difficult to keep the semantics in sync.

CUDA and CPU we have to dispatch because of cuda/no-cuda boundaries, but that doesn't apply to quantized.

Reply:
it's... complicated. Another argument for using native_functions.yaml dispatch is that it enables selective compilation easily (useful for mobile). In this particular case we don't even share preprocessing code, do we?

Reply:
sure, it's complicated, that's why I said "close to" :).

I don't believe that's the solution we've chosen for selective complication on mobile, and I'd rather avoid a complex procedure for deciding where we do the dispatch, because different people will come to different conclusions when working through the complex procedure ("well, this has only a little preprocessing code!") and we'll be left with an inconsistent mess.  "Don't do dispatch in native_functions.yaml unless it's necessary for selective complication (currently CUDA only)" seems like a good rule that can be applied consistently.

Reply:
I still added separate dispatch for `min_quant` and `max_quant` and `sort_quant` in this pr for the other native functions called by this function because they are implemented by legacy TH code.

Reply:
@gchanan - actually that's what we chose for mobile so far. I'll persuade you another time :)

Reply:
oh I see :).  In that case, go for it, but we should be consistent about it (and document the decision somewhere).  Maybe in https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/README.md ?

Reply:
will do in another diff..